driver/modern_bpf/helpers/extract/extract_from_kernel.h

```diff
@@ -57,3 +57,109 @@ static __always_inline unsigned long extract__syscall_argument(struct pt_regs *r
 
 	return arg;
 }
+
+/////////////////////////
+// GENERIC EXTRACTION
+////////////////////////
+
+/**
+ * @brief Extract `len_to_read` bytes from the pointer.
+ *
+ * The `dest` pointer usually is the stack or the auxmap.
+ * If it is the auxmap we don't have to increment the payload_pos
+ * since we are using the map as a scratch space!
+ *
+ * Please note: in case of failure the content of `dest` is not
+ * changed so we don't have to manage the return value, we have only
+ * to pass an empty value by default
+ *
+ * @param dest pointer to the destination buffer.
+ * @param len_to_read number of bytes to be read.
+ * @param src pointer to the source buffer.
+ * @return return code of `bpf_probe_read`
+ */
+static __always_inline int extract__bytebuf_from_pointer(void *dest, unsigned long len_to_read, void *src)
+{
+	return bpf_probe_read(dest, SAFE_ACCESS(len_to_read), src);
```

Comment:
Something also related to our discussion [here](https://github.com/falcosecurity/libs/pull/504#issuecomment-1195620756), for now this seems to be a user read for the open_how structure below.

Reply:
Hei @hbrueckner I will continue our discussion here :) Initially, I didn't think to support `s390x` as an official architecture, but since I have noticed your interest and effort, we will try to do it! :tada: Coming back to your question "how to do it", I think we can use a boolean like `is_kernel` and pass it to functions like `aumap__store_charbuf_param()` and `auxmap__store_bytebuf_param()`, same for `push__charbuf_param()` and `push__bytebuf_param()`... this could allow us a smooth transition and a code quite clear and simple, but let me know what you think about that :)

Reply:
This is what I had in mind https://github.com/falcosecurity/libs/pull/503/files#diff-43bc2f358f40638a28676f9ec4e1be4404c507253ba13508311cff1386d41df9R212-R227. Let me know if it sounds good to you :) 
Now it should work also on `s390x`

Reply:
Sounds good to me :) ... and I will give it a try later today.

Another thought that came into my mind was to put the address together with the type (user/kernel) into a struct to pass to the functions. And use a macro to generate it: KERNEL(addr) = { .type = KERNEL, .addr = 12345}).

Reply:
Uhm this makes sense! As the first step, I would keep things as simple as possible without hiding logic behind macros, but maybe we can do that in following steps :) @FedeDP WDYT?

Reply:
@Andreagit97 let's start w/ keeping it simple.. I scrolled thru your new commit and it looks nice... if the need arise, refactoring it should be easy towards a different prototype w/ consolidated arguments etc.

Reply:
Hi @Andreagit97 , excellent work!

Just tried [c4f28b854](https://github.com/falcosecurity/libs/pull/503/commits/c4f28b8547e9f30b3fea4a750bedaa70dc807ee8) and all tests on s390 passed (running w/o mount-point :)

/lgtm

Reply:
Thank you very much for your help!!